lib/statesman/adapters/active_record_queries.rb

```diff
@@ -66,17 +66,25 @@ def transition_name
           transition_class.table_name.to_sym
         end
 
+        def transition_reflection
+          reflect_on_association(transition_name)
+        end
+
         def model_foreign_key
-          reflect_on_association(transition_name).foreign_key
+          transition_reflection.foreign_key
+        end
+
+        def model_table
+          transition_reflection.table_name
         end
 
         def transition1_join
-          "LEFT OUTER JOIN #{transition_name} transition1
+          "LEFT OUTER JOIN #{model_table} transition1
              ON transition1.#{model_foreign_key} = #{table_name}.id"
         end
 
         def transition2_join
-          "LEFT OUTER JOIN #{transition_name} transition2
+          "LEFT OUTER JOIN #{model_table} transition2
```

Comment:
I think you need to apply this change to the new `most_recent_transition_join` method too to make this work with the `most_recent` column added in 1.2.0